File: Tete.Models/Relationships/Mentorship.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Tete.Models.Relationships
{

  public class Mentorship
  {

    [Required]
    public Guid MentorshipId { get; set; }

    [Required]
    public Guid TopicId { get; set; }

    [Required]
    public bool Active { get; set; }

    [Required]
    public DateTime CreatedDate { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    #region Learner Details

    [Required]
    public Guid LearnerUserId { get; set; }

    public string LearnerContact { get; set; }

    public bool LearnerClosed { get; set; }

    public DateTime LearnerClosedDate { get; set; }

    #endregion

    #region Mentor Details

    // A mentorship without a mentor user id is a request for a mentor.
    public Guid MentorUserId { get; set; }

    public string MentorContact { get; set; }

    public bool MentorClosed { get; set; }

    public DateTime MentorClosedDate { get; set; }

    #endregion

    public Mentorship(Guid LearnerUserId, Guid TopicId)
    {
      this.MentorshipId = Guid.NewGuid();
      this.LearnerUserId = LearnerUserId;
      this.MentorUserId = Guid.Empty;
      this.TopicId = TopicId;
      this.Active = true;
      this.CreatedDate = DateTime.UtcNow;
      this.LearnerClosed = false;
      this.MentorClosed = false;
    }
  }
}

File: Tete.Web/ClientApp/src/app/models/mentorship.ts
import { User } from "./user";

export class Mentorship {
  public mentorshipId: string;
  public learnerUserId: string;
  public learner: User;
  public learnerContact: string;
  public mentor: User;
  public mentorUserId: string;
  public mentorContact: string;
  public topicId: string;
  public active: boolean;
  public createdDate: Date;
  public startDate: Date;
  public endDate: Date;

  public learnerClosed: boolean = false;
  public learnerClosedDate: Date;

  public mentorClosed: boolean = false;
  public mentorClosedDate: Date;

  public hasMentor: boolean = false;
}